<launch>
  <arg name="planning_validator_param_path" default="$(find-pkg-share autoware_planning_validator)/config/planning_validator.param.yaml"/>
  <arg name="input_trajectory" default="/planning/scenario_planning/velocity_smoother/trajectory"/>
  <arg name="output_trajectory" default="/planning/scenario_planning/trajectory"/>
  <arg name="output_trajectory_baselink" default="/planning/scenario_planning/trajectory_baselink"/>

  <node name="planning_validator" exec="planning_validator_node" pkg="autoware_planning_validator" output="screen">
    <!-- load config a file -->
    <param from="$(var planning_validator_param_path)"/>

    <!-- remap topic name -->
    <remap from="~/input/trajectory" to="$(var input_trajectory)"/>
    <remap from="~/input/kinematics" to="/localization/kinematic_state"/>
    <remap from="~/output/trajectory" to="$(var output_trajectory)"/>
    <remap from="~/output/validation_status" to="~/validation_status"/>
  </node>


  <node name="trajectory_msg_transformer" exec="trajectory_msg_transformer_node" pkg="autoware_planning_validator" output="screen">
    <!-- set target frame id's -->
    <param name="target_frame_" value="base_link" />
    <!-- remap topic name -->
    <remap from="~/output/trajectory" to="$(var output_trajectory)"/>
    <remap from="~/output/transformed_trajectory" to="$(var output_trajectory_baselink)"/>


  </node>
  
</launch>
